beam/core/src/main/java/cz/o2/proxima/beam/direct/io/DirectBatchUnboundedSource.java

```diff
@@ -272,10 +278,18 @@ public StreamElementUnboundedReader(
       this.source = Objects.requireNonNull(source);
       this.reader = Objects.requireNonNull(reader);
       this.attributes = new ArrayList<>(Objects.requireNonNull(attributes));
-      this.toProcess = new ArrayList<>(Objects.requireNonNull(toProcess));
+      this.toProcess =
```

Comment:
Is this sort necessary? Partitions should come pre-sorted from the "sorted round-robin" added above.

Reply:
Better safe than sorry .There is virtually no cost of this.